lib/geckodriver.js

```diff
@@ -0,0 +1,194 @@
+// transpile:main
+import events from 'events';
+import _ from 'lodash';
+import { logger, util } from 'appium-support';
+import { JWProxy } from 'appium-base-driver';
+import { SubProcess } from 'teen_process';
+import { quote } from 'shell-quote';
+import { getGeckodriverForOs } from './utils';
+
+const log = logger.getLogger('Geckodriver');
+
+const LOG_DEBUG = 'debug';
+const DEFAULT_LOG_LEVEL = LOG_DEBUG;
+
+const DEFAULT_HOST = '127.0.0.1';
+const DEFAULT_PORT = 29146;
+
+const MIN_SUPPORTED_VERSION = '0.26.0';
+const FIREFOX_PACKAGE = 'org.mozilla.firefox';
+const FIREFOX_MAIN_ACTIVITY = 'org.mozilla.gecko.BrowserApp';
+
+class Geckodriver extends events.EventEmitter {
+  constructor (args = {}) {
+    super();
+
+    const {
+      host = DEFAULT_HOST,
+      port = DEFAULT_PORT,
+      geckodriverExecutable = null,
+      noReset = false,
+      debug_level = DEFAULT_LOG_LEVEL
```

Comment:
is there any reason to not use camelCase?

Reply:
nope, by bad. I'll update it.